distributed/shuffle/tests/test_shuffle.py

```diff
@@ -738,7 +739,19 @@ def __init__(self, value: int) -> None:
 
 
 @gen_cluster(client=True)
-async def test_head(c, s, a, b):
+@pytest.mark.parametrize(
+    "disk_buffer_size",
+    [
+        0,  # No in-memory buffering
+        128,  # Small enough to hit disk
+        "1GiB",  # Won't hit disk
+    ],
+)
+async def test_head(c, s, a, b, disk_buffer_size):
+    await c.run(
+        dask.config.set,
+        {"distributed.shuffle.output_max_buffer_size": disk_buffer_size},
+    )
```

Comment:
In theory all the tests could run with these parameters, so I could parametrize the whole test suite over this.